bookkeeper-server/src/main/java/org/apache/bookkeeper/discover/RegistrationManager.java

```diff
@@ -22,16 +22,36 @@
 import org.apache.bookkeeper.common.annotation.InterfaceAudience.LimitedPrivate;
 import org.apache.bookkeeper.common.annotation.InterfaceStability.Evolving;
 import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.meta.LayoutManager;
+import org.apache.bookkeeper.stats.NullStatsLogger;
 import org.apache.bookkeeper.stats.StatsLogger;
+import org.apache.bookkeeper.util.ReflectionUtils;
 import org.apache.bookkeeper.versioning.Version;
 import org.apache.bookkeeper.versioning.Versioned;
+import org.apache.commons.configuration.ConfigurationException;
 
 /**
  * Registration manager interface, which a bookie server will use to do the registration process.
  */
 @LimitedPrivate
 @Evolving
-public interface RegistrationManager extends AutoCloseable {
+public interface RegistrationManager extends LayoutManager, AutoCloseable {
```

Comment:
Why a registration manager is a layoutmanager?
They are separate concepts, aren't they?

Reply:
Thanks, will remove the extend here.